manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -282,8 +286,8 @@ func (tr *TaskReaper) tick() {
 
 // Stop stops the TaskReaper and waits for the main loop to exit.
 func (tr *TaskReaper) Stop() {
-	// TODO(dperny) calling stop on the task reaper twice will cause a panic
```

Comment:
@dperny do you remember the reason for this? I'm looking into it too.

Reply:
It's never happened, IIRC, it's just possible for it to happen. Other components have Stop channels that are guarded against double close.